File: java-bridge/src/main/java/com/xcelerate/ExcelLib.java
package com.xcelerate;

import com.sun.jna.Callback;
import com.sun.jna.Library;
import com.sun.jna.Native;
import com.sun.jna.Pointer;

interface ExcelLib extends Library {

    ExcelLib INSTANCE = Native.load(NativeLoader.resolve(), ExcelLib.class);

    interface WriteCallback extends Callback {
        void invoke(Pointer data, int length, Pointer ctx);
    }

    Pointer xcelerateOpen(
        String[]      headers,
        int           headerCount,
        int           fileType,   // 0=EXCEL, 1=CSV
        WriteCallback callback,
        Pointer       ctx
    );

    int xcelerateWrite(
        Pointer  session,
        String[] data,
        int      rowCount,
        int      colCount
    );

    int xcelerateClose(Pointer session);
}

File: java-bridge/src/main/kotlin/com/xcelerate/ExcelLib.kt
package com.xcelerate

import com.sun.jna.Library
import java.lang.annotation.Native

/**
 * JNA 브릿지 — C++ 라이브러리 함수를 직접 매핑
 */
internal interface ExcelLib : Library {

    fun generateExcel(outputPath: String, rowCount: Int): Int

    companion object {
        val INSTANCE: ExcelLib by lazy {
            Native.load("xcelerate", ExcelLib::class.java)
        }
    }
}
